### ISSUE TITLE
Feature Request:  Rewrap option so line ends at a period for LaTeX text

### ISSUE THREAD

**Issue by rickhg12hs** [stkb/Rewrap#295]
It can be useful to end each sentence with a period and then a newline in LaTeX, for example when storing the `.tex` files in a `git` repo (minimize diffs) or when just diff'ing `.tex` files in general.  This is especially useful within paragraphs.

So, using this optional feature, rewrapping a paragraph would look like this:

| Before | Rewrapped |
|----------|-----------------|
| This is a sentence.<br/> And this is another sentence that is too long.<br/>Short sentence. A sentence<br/>starting on the same line.<br/>Followed by<br/>one more. | This is a sentence.<br/> And this is another sentence <br/>that is too long.<br/>Short sentence.<br/>A sentence starting on the<br/>same line.<br/>Followed by one more. |

Currently, to replicate this functionality, the user needs to carefully select each sentence in a paragraph that already has a newline after the sentence period and then hit <kbd>Alt</kbd>+<kbd>Q</kbd>.  This is error prone, so this proposed feature request would be very useful.

**Comment by stkb** [comment]
Hi, while this is doable, I feel it's quite a specific request. So I'm reluctant to add this feature until more people request it too.

**Comment by rickhg12hs** [comment]
Understood.

For reference, amongst many others:

https://stackoverflow.com/a/6190412

**Comment by stkb** [comment]
Thanks for that.

I know it's not everything you want, but would it be ok if it simply preserved existing newlines after a '.'? So you have to make sure you write one-sentence-per-line yourself, but then Rewrap will keep that:

| Before | Rewrapped |
|----------|-----------------|
| This is a sentence.<br/> And this is another sentence that is too long.<br/>Short sentence.<br/> A sentence<br/>starting on the next line.<br/>Followed by<br/>one more. | This is a sentence.<br/> And this is another sentence <br/>that is too long.<br/>Short sentence.<br/>A sentence starting on the<br/>next line.<br/>Followed by one more. |

I can easily roll that into existing settings. Also trying to detect sentence endings within lines by `.`'s can be fraught with false positives.

**Comment by rickhg12hs** [comment]
If I understand correctly, yes your proposal would be helpful.  Currently I need to carefully select each line a sentence is on.  With your proposal, I could just select the entire paragraph, yes?

**Comment by stkb** [comment]
Yeah, so if you had the input on the left here and just pressed alt+q while the cursor was in that paragraph, you'd get the output on the right.

![Sentences](https://user-images.githubusercontent.com/6125444/143880883-d3209a78-8bfb-4597-bf71-d30dcd41d4a1.png)

The only thing it doesn't do is "fix" the `Short sentence. A sentence...`. So it would be up to you to make sure that every sentence starts on a new line, but then that would be preserved.

**Comment by rickhg12hs** [comment]
Ah, I see.  It's not perfect, but it would be very helpful.  It's almost muscle memory for me now to hit <kbd>Enter</kbd> after finishing a sentence.  Rewrapping becomes the most useful during the revising stage - As you can imagine, the length of lines change a lot during insert/delete/modify editing.

**Comment by stkb** [comment]
FYI Having all sentences reformatted (thing I said I probably wouldn't do) is looking more likely now it's also been requested in #301.

**Comment by dhimmel** [comment]
I'd also like an option to rewrap not based on column width, but by punctuation such that the result is one sentence/phrase per line.

This is what [we suggest](https://github.com/manubot/rootstock/blob/9504aa7751e9d5ca0e7df896bf3aeeb73d2fb262/USAGE.md#manubot-markdown) for writing in markdown with Manubot:

> However, "one paragraph per line" makes the git diff less precise, leading to less granular review commenting, and makes conflicts more likely. Therefore, we recommend using [semantic linefeeds](https://rhodesmill.org/brandon/2012/one-sentence-per-line/) — newlines between sentences. We have found that "one sentence per line" is preferable to "word wrap" or "one paragraph per line".

Would be nice to have this automated. When tracking prose with git, one sentence/phrase per line creates the cleanest diffs.

**Comment by reagle** [comment]
Semantic line-feeds reflow content so that single line-breaks follow punctuation, and double-line breaks signify paragraphs; this is also known as [semantic line breaks](https://sembr.org/). I wrote a [plugin for Sublime Text](https://gist.github.com/reagle/3457fb5ef94e898f47777cb4065aa1f1) that does this.

**Comment by nstarman** [comment]
Chiming in to also request a one-sentence-per-line wrapping for latex.

**Comment by straight-shoota** [comment]
I'd very much appreciate it, if line breaks after a sentence could be preserved.
I figure this could be an alternative mode for `doubleSentenceSpacing` with the options: single space, double space, newline.
